usefulUtilities/hifiPing/webApp/app/pingServer.js

```diff
@@ -0,0 +1,325 @@
+                status: "error",
+                text: "Error while deleting subscription from DB! " + JSON.stringify(error)
+            };
+
+            response.statusCode = 500;
+            response.setHeader('Content-Type', 'application/json');
+            return response.end(JSON.stringify(responseObject));
+        }
+        
+        var responseObject = {
+            status: "success"
+        };
+
+        response.statusCode = 410;
+        response.setHeader('Content-Type', 'application/json');
+        return response.end(JSON.stringify(responseObject));
+    });
+}
+
+
+// Uses the web-push library to send an encrypted push notification to the user's browser
+function sendNotification(username, subscription, payloadText, response) {
+    return webpush.sendNotification(subscription, payloadText)
+        .then(() => {
+            var responseObject = {
+                status: "success"
+            };
+
+            response.statusCode = 200;
+            response.setHeader('Content-Type', 'application/json');
+            return response.end(JSON.stringify(responseObject));
+        })
+        .catch((err) => {
+            console.log(JSON.stringify(err));
+            if (err.statusCode === 410) {
+                deleteSubscriptionFromDatabase(username, response)
+            } else {
+                var responseObject = {
+                    status: "error",
+                    text: "Subscription is no longer valid."
+                };
+    
+                response.statusCode = 500;
+                response.setHeader('Content-Type', 'application/json');
+                return response.end(JSON.stringify(responseObject));
+            }
+        });
+}
+
+
+// Gets a user's push notification subscription information from the database
+function getSubscriptionFromDatabase(targetUsername, senderDisplayName, senderHref, response) {
+    var query = `SELECT * FROM \`subscriptions\` WHERE username='${targetUsername}'`;
+
+    connection.query(query, function (error, results) {
+        if (error) {
+            var responseObject = {
+                status: "error",
+                text: "Error while retrieving subscription! " + JSON.stringify(error)
+            };
+
+            response.statusCode = 500;
+            response.setHeader('Content-Type', 'application/json');
+            return response.end(JSON.stringify(responseObject));
+        }
+
+        // We don't want anyone to know that the server doesn't have that username in the DB
+        if (results.length === 0) {
+            console.log("User tried to send a notification to someone who isn't in the DB: " + targetUsername);
+            var responseObject = {
+                status: "success"
+            };
+
+            response.statusCode = 200;
+            response.setHeader('Content-Type', 'application/json');
+            return response.end(JSON.stringify(responseObject));
+        } else if (results.length > 1) {

```

Comment:
This should absolutely never happen, because the username is the primary key in the MySQL table. If it does, something is seriously wrong, and I'll have other things to worry about.